From: Jan Kara <jack@suse.cz>
Subject: loop: Reintroduce lo_ctl_mutex removed by commit 310ca162d
References: bsc#1124974
Patch-mainline: Never, kabi

Commit 310ca162d "block/loop: Use global lock for ioctl() operation." removed
lo_ctl_mutex as it is not needed anymore. No external module has any business
in touching it (the whole structure is exported for cryptoloop module) but
let's reintroduce the mutex to avoid kABI breakage.

Signed-off-by: Jan Kara <jack@suse.cz>

---
 drivers/block/loop.h |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/block/loop.h
+++ b/drivers/block/loop.h
@@ -56,6 +56,7 @@ struct loop_device {
 
 	spinlock_t		lo_lock;
 	int			lo_state;
+	struct mutex		lo_ctl_mutex;
 	struct kthread_worker	worker;
 	struct task_struct	*worker_task;
 	bool			use_dio;
